docs: update docs to reflect production mode by default

alan-agius4 opened the pull request:
In CLI version 12, we introduced the concept of production builds by default.

With this change we update the documentation to reflect the changes.

More information about the change can be found https://github.com/angular/angular-cli/pull/20128

aikithoughts reviewed (commented):
Looking good! A few edits to help with readability.

alan-agius4 committed:
docs: update docs to reflect production mode by default

In CLI version 12, we introduced the concept of production builds by default.

With this change we update the documentation to reflect the changes.

More information about the change can be found https://github.com/angular/angular-cli/pull/20128

alan-agius4 commented on aio/content/guide/workspace-config.md:
```diff
@@ -174,16 +174,14 @@ The `architect/build` section configures defaults for options of the `ng build`
 ### Alternate build configurations
 
-By default, a `production` configuration is defined, and the `ng build` command has `--prod` option that builds using this configuration. The `production` configuration sets defaults that optimize the app in a number of ways, such as bundling files, minimizing excess whitespace, removing comments and dead code, and rewriting code to use short, cryptic names ("minification").
+By default, a `production` and `development` configurations are defined, the `ng build` command is configurated to run with the `production` configuration. The `production` configuration sets defaults that optimize the app in a number of ways, such as bundling files, minimizing excess whitespace, removing comments and dead code, and rewriting code to use short, mangled names ("minification").
```
Much better thanks.

mary-poppins commented:
You can preview 943e91d at https://pr41173-943e91d.ngbuilds.io/.

mary-poppins commented:
You can preview 72aeb52 at https://pr41173-72aeb52.ngbuilds.io/.

aikidave commented on aio/content/guide/glossary.md:
```diff
@@ -907,7 +907,7 @@ A buildable or runnable subset of a [project](#project), configured as an object
 
 In the `angular.json` file, each project has an "architect" section that contains targets which configure builders. Some of these targets correspond to [CLI commands](#cli), such as `build`, `serve`, `test`, and `lint`.
 
-For example, the Architect builder invoked by the `ng build` command to compile a project uses a particular build tool, and has a default configuration whose values can be overridden on the command line. The `build` target also defines an alternate configuration for a "production" build, that can be invoked with the `--prod` flag on the `build` command.
+For example, the Architect builder invoked by the `ng build` command to compile a project uses a particular build tool, and has a default configuration with values that you can override on the command line. The `build` target also defines an alternate configuration for a "development" build, which you can invoke with the `--configuration development` flag on the `build` command.
```
Just a minor nit that I didn't catch before. It's a best practice to avoid using quotation marks for emphasis. So in this case:

...also defines an alternate configuration for a development build, which you can...

aikidave commented on aio/content/guide/workspace-config.md:
```diff
@@ -174,16 +174,18 @@ The `architect/build` section configures defaults for options of the `ng build`
+* Bundling files
+* Minimizing excess whitespace
+* Removing comments and dead code
+* Rewriting code to use short, mangled names (minification).
```
```suggestion
* Rewriting code to use short, mangled names (minification)
```
No need for a period in this case, as the other items in the list aren't complete sentences either.

aikithoughts reviewed (approved):
A few little items, but otherwise LGTM!

aikithoughts reviewed (approved):
Reviewed-for: global-docs-approvers

jelbourn commented on aio/content/guide/deployment.md:
```diff
@@ -68,7 +68,7 @@ ng deploy
 The command is interactive. In this case, you must have or create a Firebase account, and authenticate using that account. The command prompts you to select a Firebase project for deployment
 
-After the command produces an optimal build of your application (equivalent to `ng deploy --prod`), it'll upload the production assets to Firebase.
+After the command produces an optimal build of your application (equivalent to `ng deploy`), it uploads the production assets to Firebase.
```
```suggestion
The command builds your application and uploads the production assets to Firebase.
```
I started commenting to say that the bit in parenthesis could be dropped now, but this whole sentence can be shortened a bit.

jelbourn commented on aio/content/guide/deployment.md:
```diff
@@ -325,7 +325,7 @@ In addition to build optimizations, Angular also has a runtime production mode.
 
 Switching to _production mode_ makes it run faster by disabling development specific checks such as the dual change detection cycles.
 
-When you enable production builds via `--prod` command line flag, the runtime production mode is enabled as well.
+When you enable production builds, the runtime production mode is enabled as well.
```
I would move this sentence into the previous paragraph like
```markdown
_Production mode_ improves application performance by disabling development-only safety
checks and debugging utilities, such as the expression-changed-after-checked detection.
Building your application with the production configuration automatically enables Angular's
runtime production mode.
```

jelbourn commented on aio/content/guide/workspace-config.md:
```diff
@@ -174,16 +174,18 @@ The `architect/build` section configures defaults for options of the `ng build`
 ### Alternate build configurations
 
-By default, a `production` configuration is defined, and the `ng build` command has `--prod` option that builds using this configuration. The `production` configuration sets defaults that optimize the app in a number of ways, such as bundling files, minimizing excess whitespace, removing comments and dead code, and rewriting code to use short, cryptic names ("minification").
+By default, both a `production` and `development` configurations are defined. The `ng build` command is configured to run with the `production` configuration. This configuration sets defaults that optimize the app in a number of ways, such as:
```
```suggestion
Angular CLI comes with two build configurations: `production` and `development`. By default, the `ng build` command uses the `production` configuration, which applies a number of build optimizations, including:
```
More active voice

jelbourn commented on aio/content/start/start-deployment.md:
```diff
@@ -51,10 +51,10 @@ A best practice is to run your project locally before you deploy it. To run your
 ## Building and hosting your application
 
- 1. To build your application for production, use the `build` command with the `prod` flag.
+ 1. To build your application for production, use the `build` command. By default, this command is configured to `production` mode.
```
```suggestion
 1. To build your application for production, use the `build` command. By default, this command uses the `production` build configuration.
```
more active voice

jessicajaniuk reviewed (approved):
LGTM 🍪 

reviewed-for: public-api

alan-agius4 committed:
fixup! docs: update docs to reflect production mode by default

alan-agius4 commented:
@jelbourn comments addressed. PTAL

mary-poppins commented:
You can preview 89e958b at https://pr41173-89e958b.ngbuilds.io/.

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api

petebacondarwin reviewed (commented):
Does the `ng test` task also default to the `production` configuration?

petebacondarwin reviewed (approved):
Reviewed-for: public-api

alan-agius4 commented:
@petebacondarwin 

> Does the ng test task also default to the production configuration?

No, `ng test` has no development/production configuration
